templates/csharp/SmartContract1.cs

```diff
@@ -10,8 +12,30 @@
 
 public class SmartContract1 : SmartContract
 {
-    public static void Main()
+    public static object Main(string method, object[] args)
+    {
+        if (method == "PutValue") 
+        {
+            return PutValue(args[0] as byte[]);
+        } 
+        else if (method == "GetValue")
+        {
+            return GetValue();
+        }
+        else 
+        {
+            throw new NotImplementedException();
```

Comment:
try/catch support comming in neovm3m but not supported on neovm2. Typically, Main method returns false when method name doesn't match. Examples: https://github.com/neo-ngd/CNEO-Contract/blob/master/NeoContract/CNEO.cs#L109, https://github.com/Moonlight-io/moonlight-ico-template/blob/master/ICOTemplate/ICOTemplate.cs#L185